src/gt4py/next/program_processors/runners/dace_iterator/utility.py

```diff
@@ -181,6 +181,24 @@ def unique_var_name():
     return unique_name("_var")
 
 
+def map_field_dimensions_to_sdfg_symbols(
+    name: str,
+    dims: Sequence[Dimension],
+    neighbor_tables: Mapping[str, NeighborTable],
+    sort_dims: bool,
+) -> tuple[list[dace.symbol], list[dace.symbol]]:
+    dtype = dace.int64
+    sorted_dims = [dim for _, dim in get_sorted_dims(dims)] if sort_dims else dims
+    shape = [
+        neighbor_tables[dim.value].max_neighbors
+        if dim.kind == DimensionKind.LOCAL
+        else dace.symbol(unique_name(f"{name}_shape{i}"), dtype)
+        for i, dim in enumerate(sorted_dims)
+    ]
+    strides = [dace.symbol(unique_name(f"{name}_stride{i}"), dtype) for i, _ in enumerate(shape)]
+    return shape, strides
+
+
 def new_array_symbols(name: str, ndim: int) -> tuple[list[dace.symbol], list[dace.symbol]]:
```

Comment:
I will keep it for now, because it is still used when we allocate arrays (for both fields and connectivities) in nested scope. The constant dimensions are propagated through the levels of nesting by the simplify pass.

Reply:
This make sense.
However, I do not like this "array creation in the wild", there should be probably something like `add_storage()`.